content/posts/2020-03-31-continuous-stuff-with-github-actions/index.mdx

```diff
@@ -0,0 +1,256 @@
+
+## Manual workflows with comments
+
+We can also react to comments on the Pull Request itself, which can be useful if you want a more seamless integration of
+some features.
+
+In my case, I wanted to deploy a preview version of Pull Requests to our frontend application.
+It wasn't efficient to do this for every single PR though (only a small subset of them actually need this), so we went
+with this instead:
+
+![Comments triggering GitHub Action](./github-actions-comment.png)
+
+The ease with which this was all done by just using different hooks made this very pleasant to work with.
+
+
+# Reusable actions
+
+A second issue I often had with previous CIs was the difficulty, or complete lack of a way to create reusable parts of
+your pipeline, so you can compose other jobs with them.
+
+Yes, yaml has some syntax that allows you to reuse blocks...
+
+```yaml
+
+component: &compoenent
+  foo: bar
+
+extension: *component
+  baz: biz
+```
+
+... but they don't look pretty, especially once they start to grow.
+And if you, like me, have any experience mantaining a large project with multiple CI
+worflows and configurations, you probably know that things tend to get out of hand.
+It's always a single YAML file, which can grow to hundreds of lines. You can reuse blocks of YAML, but they may end up
+running under different contexts (e.g.: different docker images, different dependencies installed, etc).
+
+CircleCI did introduce the concept of [Orbs][circleci-orbs] in their ecosystem, which attempts to tackle this. However,
+their reusability is limited.
+
+```yaml
+# circleci.yml
+
+push_s3:
+  executor: ubuntu
+  steps:
+    - my-custom-org/my-custom-command:
+        arg: "foo"
+```
+
+This is a trimmed-down example, where a custom command is encapsulated in a `my-custom-orb` Orb. You may notice that the
+main config file is the one who specificies the execution environment (ubuntu, in this case).
+So, if the Orb tries to `yum install git`, this would fail, because that package manager isn't used in Ubuntu.
+
+So you end up with a mess of a script that does a whole bunch of magic just to figure out how to install the
+dependencies it needs. Check out [the actual source code](orb-awscli) for the official Orb that installs `awscli` on your
+jobs. It's a bit of a mess, isn't it?
+
+You could delegate everything into a ready-to-go Docker container, and run whatever you need in there. But then you lose
+access to the overall filesystem of your original job, where you already cloned your project and created a bunch of
+useful artifacts. Not great, either.
+
+GitHub Actions solves this by allowing you to write reusable actions in two different ways. None of those are YAML, and
+both of those are way better:
+
+1. As a docker container. But one that will automatically mount the local filesystem into it, so that you get access to
+   everything your job has been doing so far. And since, it's docker, you can pretty much install whatever dependencies
```

Comment:
```suggestion
1. As a Docker container, but one that will automatically mount the local filesystem into it. You get access to
   everything your job has been doing so far, and, since it's Docker, you can pretty much install whatever dependencies
```